### PR TITLE
CTDA-667: Test Support - populate MesRecMES6 field

### PR DESCRIPTION
**Pull request by jhkldev**

### CODE REVIEW

File: app/controllers/ArrivalTestMessagesController.scala [hunk 1]
```diff
@@ -50,26 +52,38 @@ class ArrivalTestMessagesController @Inject()(cc: ControllerComponents,
...
-                        postResponse.status match {
-                          case status if is2xx(status) =>
-                            Created
-                          case _ =>
-                            handleNon2xx(postResponse)
-                        }
+                    .getMessages(arrivalId)
+                    .flatMap {
+                      case Right(arrivalWithMessages: ArrivalWithMessages) =>
+                        val message = populateMesRecMES6(request.generatedMessage, "IE007", arrivalWithMessages.messages)
```
Comment (marcinkrykowski) [inline app/controllers/ArrivalTestMessagesController.scala:58]: Maybe we can extract that "IE007" to be a separate constant?

File: app/controllers/DepartureTestMessagesController.scala [hunk 2]
```diff
@@ -50,15 +52,27 @@ class DepartureTestMessagesController @Inject()(cc: ControllerComponents,
...
-                        postResponse.status match {
-                          case status if is2xx(status) =>
-                            Created
-                          case _ =>
-                            handleNon2xx(postResponse)
-                        }
+                    .getMessages(departureId)
+                    .flatMap {
+                      case Right(departureWithMessages: DepartureWithMessages) =>
+                        val message = populateMesRecMES6(request.generatedMessage, "IE015", departureWithMessages.messages)
```
Comment (marcinkrykowski) [inline app/controllers/DepartureTestMessagesController.scala:58]: A similar extracting case as the one above

**Review by marcinkrykowski — CHANGES_REQUESTED** [review]

**Comment by RobertLaverick** [comment]
Are we sure that we're going to have the field there but empty? From the look of the code we don't add the field, we expect it to be there and blank, what will this do if the field is empty?

Also, have you run the produced xml through the validator?

File: app/controllers/DepartureTestMessagesController.scala (hunk 2 above)
Comment (jhkldev) [reply app/controllers/DepartureTestMessagesController.scala:58]: I'll add MessageType declarations to address this.

**Comment by jhkldev** [comment]
> Are we sure that we're going to have the field there but empty? From the look of the code we don't add the field, we expect it to be there and blank, what will this do if the field is empty?
> 
> Also, have you run the produced xml through the validator?



> Are we sure that we're going to have the field there but empty? From the look of the code we don't add the field, we expect it to be there and blank, what will this do if the field is empty?
> 
> Also, have you run the produced xml through the validator?

The code assumes that the MesSenMES3 field will be present in both IE007 and IE015. The field has to be added with a non empty value retrieved from SynVerNumMES2 in the POST submission.

There's also an assumption that MesSenMES3 and MesRecMES6 fields exist in any EIS responses as these are mandatory fields (as specified in the schema).

I can add tests to make sure the generated XML passes XSD validation.

**Comment by jhkldev** [comment]
Closing PR as the MesRecMES6 field does not need to be populated.

**Pull request closed without merging**